### PR TITLE
[Resource][Data Source] ACR Scope Map and Tokens

### PR DESCRIPTION
**Pull request by Lucretius**
- Data Source for `azurerm_container_registry_token`
- Data Source for `azurerm_container_registry_scope_map`
- Resource for `azurerm_container_registry_token`
- Resource for `azurerm_container_registry_scope_map`
- Documentation for each
- Tests for each, they all pass running against my own Azure environment.

There are passwords for the token which I would like to expose but there is no way to delete them once generated so they didn't really fit into the lifecycle management of the token itself.  Perhaps they might be better suited as a separate data source, similar to how we handle the SAS tokens?

Also note that these are still both preview features.  As such I've vendor-included the SDK preview code and only used it in these specific preview features (see `client.go`)

This PR resolves https://github.com/terraform-providers/terraform-provider-azurerm/issues/6505

### CODE REVIEW

**Comment by shashken** [comment]
@Lucretius @tombuildsstuff Any updates on this PR?

File: website/docs/r/container_registry_token.html.markdown [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+  Manages an Azure Container Registry token.
+
+---
+
+# azurerm_container_registry_token
+
+Manages an Azure Container Registry token.  Tokens are a preview feature only available in Premium SKU Container registries.
+
+```hcl
+resource "azurerm_resource_group" "rg" {
```
Comment (katbyte) [inline website/docs/r/container_registry_token.html.markdown:15]: could we use example for all names
```suggestion
resource "azurerm_resource_group" "example" {
```

File: website/docs/r/container_registry_token.html.markdown [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+  name     = "resourceGroup1"
```
Comment (katbyte) [inline website/docs/r/container_registry_token.html.markdown:16]:
```suggestion
  name     = "example-resource-group"
```

File: website/docs/r/container_registry_token.html.markdown [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+  location = "West US"
+}
+
+resource "azurerm_container_registry" "acr" {
+  name                     = "containerRegistry1"
```
Comment (katbyte) [inline website/docs/r/container_registry_token.html.markdown:21]:
```suggestion
  name                     = "ExampleRegistry"
```

and so forth

**Review by katbyte — CHANGES_REQUESTED** [review]
Hi @Lucretius - sorry for the delay in reviewing this, overall it looks good but we'll need to fix up the merge conflicts and move to the new testing framework. Once thats done this should be good to merge!

**Comment by katbyte** [comment]
closing as this PR has been superseded by #11350

**Pull request closed without merging**